kube-runtime/src/controller/mod.rs

```diff
@@ -122,6 +122,23 @@ where
     })
 }
 
+pub fn trigger_self_shared<K, S>(
```

Comment:
maybe leave this private for now, have some ideas on how to expose these triggers better - will write a ticket